lms/djangoapps/verify_student/views.py

```diff
@@ -492,6 +487,13 @@ def _redirect_if_necessary(
                     url = reverse('verify_student_verify_now', kwargs=course_kwargs)
             else:
                 url = reverse('verify_student_start_flow', kwargs=course_kwargs)
+        elif message in [self.FIRST_TIME_VERIFY_MSG, self.UPGRADE_MSG]:
```

Comment:
`UPGRADE_MSG` would never factor in here since we are already matching on it in the previous elif block above. Are you sure this is going to work the way you expect?